File: sod/sod-monitor-center/sod-monitor-calculate/src/main/scala/com/piesat/calculate/entity/task/DiTaskConfiguration.scala
package com.piesat.calculate.entity.task

import java.util.Date

import com.alibaba.fastjson.annotation.JSONField
import com.fasterxml.jackson.annotation.{JsonFormat, JsonProperty}
import com.piesat.calculate.entity.BaseEntity

import scala.beans.BeanProperty

/**
  * Created by zzj on 2020/4/3.
  */
class DiTaskConfiguration extends BaseEntity{
  /**
    * 任务编号
    */
  @BeanProperty
  var taskId: String = null
  /**
    * 任务所属系统
    */
  @BeanProperty
  var system:String = null
  /**
    * 任务名称
    */
  @BeanProperty
  var taskName:String = null
  /**
    * 任务类型 1.清除,2.备份,3.迁移
    */
  @BeanProperty
  var taskDuty:String = null
  /**
    * cron表达式
    */
  @BeanProperty
  var taskCron:String = null
  /**
    * 超时时间 分钟
    */
  @BeanProperty
  var overtime:Int = 0
  /**
    *重试次数
    */
  @BeanProperty
  var taskMaxTime:Int = 0

  /**
    * 任务执行时间偏移量
    */
  @BeanProperty
  var offset:Int = 0
  /**
    * 告警时间间隔
    */
  @BeanProperty
  var alarmtime:Int = 0

  /**
    * 是否告警 0 不告警 1告警
    */
  @BeanProperty
  var isAlarm:Int = 0

  /**
    * 四级编码
    */
  @BeanProperty
  var dataType:String = null

  /**
    * 物理库名称
    */
  @BeanProperty
  @JsonProperty("SEND_PHYS")
  var sendPhys:String = null

  /**
    * 环节
    */
  @BeanProperty
  var handleProcess:Int = 0
  /**
    * 样本个树
    */
  @BeanProperty
  var src:Int = 0
  /**
    * 算法名称
    */
  @BeanProperty
  @JsonProperty("PROD_SYS")
  var prodSys:String = null

  @BeanProperty
  var isDelete:Int = 0

  @BeanProperty
  var id: String = _

  @BeanProperty
  @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss:SSS", timezone = "GMT+8")
  var ddateTime: Date = _
}

File: sod/sod-schedule-center/sod-schedule-rpc-api/src/main/java/com/piesat/schedule/rpc/vo/DiTaskConfiguration.java
package com.piesat.schedule.rpc.vo;

import com.alibaba.fastjson.annotation.JSONField;

/**
 * @program: DataStoragePlatform
 * @描述
 * @创建人 zzj
 * @创建时间 2019/4/11 9:24
 */
public class DiTaskConfiguration {
    private String taskId;
    private String system;
    private String taskName;
    private String taskDuty;
    private String taskCron;
    private int overtime;
    private int taskMaxTime;
    private int offset;
    private int alarmtime;
    private int isAlarm;
    private String dataType;
    @JSONField(name= "SEND_PHYS")
    private String sendPhys;
    private int handleProcess;
    private int src;
    @JSONField(name= "PROD_SYS")
    private String prodSys;



    public String getTaskId() {
        return taskId;
    }

    public void setTaskId(String taskId) {
        this.taskId = taskId;
    }

    public String getSystem() {
        return system;
    }

    public void setSystem(String system) {
        this.system = system;
    }

    public String getTaskName() {
        return taskName;
    }

    public void setTaskName(String taskName) {
        this.taskName = taskName;
    }

    public String getTaskDuty() {
        return taskDuty;
    }

    public void setTaskDuty(String taskDuty) {
        this.taskDuty = taskDuty;
    }

    public String getTaskCron() {
        return taskCron;
    }

    public void setTaskCron(String taskCron) {
        this.taskCron = taskCron;
    }

    public int getOvertime() {
        return overtime;
    }

    public void setOvertime(int overtime) {
        this.overtime = overtime;
    }

    public int getTaskMaxTime() {
        return taskMaxTime;
    }

    public void setTaskMaxTime(int taskMaxTime) {
        this.taskMaxTime = taskMaxTime;
    }

    public int getOffset() {
        return offset;
    }

    public void setOffset(int offset) {
        this.offset = offset;
    }

    public int getAlarmtime() {
        return alarmtime;
    }

    public void setAlarmtime(int alarmtime) {
        this.alarmtime = alarmtime;
    }

    public int getIsAlarm() {
        return isAlarm;
    }

    public void setIsAlarm(int isAlarm) {
        this.isAlarm = isAlarm;
    }

    public String getDataType() {
        return dataType;
    }

    public void setDataType(String dataType) {
        this.dataType = dataType;
    }


    public int getHandleProcess() {
        return handleProcess;
    }

    public void setHandleProcess(int handleProcess) {
        this.handleProcess = handleProcess;
    }

    public int getSrc() {
        return src;
    }

    public void setSrc(int src) {
        this.src = src;
    }

    public String getSendPhys() {
        return sendPhys;
    }

    public void setSendPhys(String sendPhys) {
        this.sendPhys = sendPhys;
    }

    public String getProdSys() {
        return prodSys;
    }

    public void setProdSys(String prodSys) {
        this.prodSys = prodSys;
    }
}
